build_automation/frontend/src/js/main_screen.js

```diff
@@ -26,8 +40,39 @@ class MainScreen extends React.Component {
         this.setState({ currentTab: selectedTab });
       };
 
+    componentDidMount() {
+        this.showBadge();
+        this.timerID = setInterval(
+            () => this.showBadge(),1000*60*60
+        );
+    }
+
+    componentWillUnmount() {
+        clearInterval(this.timerID);
+    }
+
+    showBadge() {
+        axios
+            .get(APP_URLS.DISKSPACE, {responseType: 'json'})
+            .then((response) => {
+                this.used = 100*(response.data.total_space-response.data.available_space)/response.data.total_space
```

Comment:
Is `this` bound to the current instance within the axios's successful promise resolution?